Update Amazee docker images upon pygmy update

fubarhouse opened the pull request:
Closes #20 by introducing image updates based upon the original issue notes.

It will update the images matching 'amazee/*' when `pygmy update` is called, and it will instead treat each one independently, complete with error handling for each available image as well as checks for empty lists.

If the solution is not idea, I would like to work with you to improve it so please provide any feedback you can.

fubarhouse committed:
#20: Introduce automatic updates to amazee/* images

Adds an Amazee class which upon being called via pull_all will gather a list of images matching ‘amazee/‘ and proceed to pull the images independently with error handling consistent with other parts of pygmy.

fubarhouse committed:
Merge branch 'master' of github.com:fubarhouse/pygmy into issue/20

fubarhouse committed:
#20: Improve error handling/reporting of list command

fubarhouse committed:
#20: Prevent the inclusion of haproxy and ssh-agent from the docker images being included in this logic

Schnitzel commented:
wow @fubarhouse that's awesome, thank you! Assigning to @dasrecht so he can have a look

fubarhouse commented:
I've noticed `pygmy update` without any running amazee images causes an error so I'll get a fix in place tonight. It is in fact an edge case which most people shouldn't fit, but I'd like it to be remediated sooner rather than later.

fubarhouse committed:
#20: Prevent false failure of docker command on amazee image list command when updates are requested and none are available.

This ensures the proceeding logic can process the result regardless of output by providing an exit code of 0. Output will be either nothing or the relevant information separated by '\n', and we can safely prevent falsly failing on these edge cases.

fubarhouse committed:
Rework amazee image updates for windows compatibility

fubarhouse committed:
Merge pull request #1 from fubarhouse/comparrison-improvements

Rework amazee image updates for windows compatibility

dasrecht commented:
Hi @fubarhouse  thanks for the work!
Reviewed and tested. Will merge it into master and roll it into the next pygmy release.
